### ISSUE TITLE
World Block State and Inventories

### ISSUE THREAD

**Issue by EliteSpectre** [maruohon/litematica#600]
I cant seem to get the "full" World Block State/blockInfoOverlay Thing to work, specifically when looking at schematics. I want to save Chest and various other inventories in my schematics or if available, see them from pre made schematics by other people. I don't know if I missed some feature I'd have to enable for this to save but when saving in singleplayer, inventories don't get saved in the schematic. I even looked a bit into Fallen Breaths tweakermore mod, in which I found some syncing feature between server and so on. In its description it says, that it'll sync block entities & data during schematic saving, which still has no effect. I'm currently working on a storage project and being able to see what items should be in hoppers, for example, is kind of necessary.

**Comment by maruohon** [comment]
Inventories (and basically any block entity or entity NBT data) are supposed to get saved from single player worlds (and always have been).

So you have some blocks that have some stuff in their inventory contents in a single player world, and you save a schematic of those. But the Block Info Overlay shows an empty inventory? And also if you paste it in single player, the pasted blocks also don't have their contents anymore? Can you send me one such schematic? Also do you have lots of other mods, have you tried if it works the same without any other mods (than my client mods)? And which MC and mod version are you using?

**Comment by EliteSpectre** [comment]
Exactly, I've investigated a bit beforehand and found out that saving schematics on servers isn't possible without Servux since data only gets transferred as long as you have the inventory of blocks open. Though in singleplayer it should work as I understand, but as you said even when saved in singleplayer and pasted in the same instance, blocks don't have their inventory filled. The BlockInfoOverlay only seems to work when looking at actual blocks I placed and none which should've been saved to the schematic nor after pasting. As for schematics themselves, it doesn't matter because all of them don't work. I am currently playing in 1.19.2 Fabric, specifically the Fabulously Optimized modpack in the curseforge launcher. Aside from the mods that come with it, I also added some personalized mods like your collection and world maps etc. Now of course it could be one of them if I hadn't tested the mods folder of cubicmetre (his mods were on 1.18.2 and I just updated them to 1.19.2 if there was an update available), which he linked in his Greatest Storage Systems Video 3 months ago. In his video he even briefly mentioned (11:29) that you can create your own filter quickly and use the block info overlay to see what you have to put in the hoppers. Regarding version of the mod, I tried the official curseforge version (1.19.2-0.12.4) and the "unreleased" (1.19.2-0.12.6-pre.1). Both of them didn't work. Could some feature or command from your mods and mostly the carpet "pack" (carpet fixes, carpet tis addition and carpet itself) influence this feature?

**Comment by EliteSpectre** [comment]
Okay so after a bunch of testing with mods I tried to completely uninstall Minecraft and reinstall all the mods. It finally works with ALL the mods concluding that they didn't interfere with it, but I think the problem was some kind of setting or feature of some mod that might've broke this feature. Since I experimented with some "invasive" mods before, it very well could've been them as well, even after removing them.

**Comment by EliteSpectre** [comment]
Okay so after a bunch of testing with mods I tried to completely uninstall Minecraft and reinstall all the mods. It finally works with ALL the mods concluding that they didn't interfere with it, but I think the problem was some kind of setting or feature of some mod that might've broke this feature. Since I experimented with some "invasive" mods before, it very well could've been them as well, even after removing them.
